builtin/credential/aws/path_login.go

```diff
@@ -612,6 +613,9 @@ func (b *backend) pathLoginUpdateEc2(ctx context.Context, req *logical.Request,
 	if err != nil {
 		return nil, err
 	}
+	if !cidrutil.RemoteAddrIsOk(req.Connection.RemoteAddr, roleEntry.BoundCIDRs) {
+		return logical.ErrorResponse("login request originated from invalid CIDR"), nil
```

Comment:
I think these checks need to be removed. The reason is that we have use-cases where people want to generate tokens for different sets of CIDRs than where they're coming from. In the most specific sense the right way to do this is token store roles, and it likely isn't really the same issue when talking about machine auth, but -- the other way to look at it is, they can get a token, but they can't actually use it since if they try to use it from the wrong CIDR it'll be denied.

So I think here this could be considered defense in depth but I just wonder if it's going to be too onerous.

Reply:
I'm actually seeing that we check this elsewhere. Probably we need to plumb this everywhere or remove it from the places that already have it.